### PR TITLE
Build ImageChangeController shouldn't return early if it encounters an Instantiate error

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: #2595

### CODE REVIEW

**Comment by mfojtik** [comment]
@ncdc PTAL (i will rebase the first commit as soon as https://github.com/openshift/origin/pull/2620 will be merged)

File: pkg/build/controller/image_change_controller.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ import (
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
+var BuildConfigInstantiateError = fmt.Errorf("not all build configurations processed successfully, will retry")
```
Comment (ncdc) [inline pkg/build/controller/image_change_controller.go]: Use `errors.new`. The error message could probably be a bit more descriptive, maybe something like: `an error occurred processing 1 or more build configurations; the image change trigger for image stream $ns/$stream:$tag will be retried`

**Comment by mfojtik** [comment]
@ncdc fixed, PTAL

File: pkg/build/controller/image_change_controller.go [hunk 2]
```diff
@@ -110,22 +118,24 @@ func (c *ImageChangeController) HandleImageRepo(repo *imageapi.ImageStream) erro
...
 				}
-
-				return fmt.Errorf("error instantiating Build from BuildConfig %s/%s: %v", config.Namespace, config.Name, err)
+				hasError = true
+				continue
 			}
 		}
 	}
+	if hasError {
+		return errors.New(fmt.Sprintf("an error occurred processing 1 or more build configurations; the image change trigger for image stream %s/%s:%s will be retried", repo.Namespace, repo.Status.DockerImageRepository))
```
Comment (ncdc) [inline pkg/build/controller/image_change_controller.go]: :smile: fmt.Errorf :smile: 

Also, the varargs don't match the text

Comment (mfojtik) [inline pkg/build/controller/image_change_controller.go]: @ncdc I knew you will not like the errors.New :D

Comment (ncdc) [inline pkg/build/controller/image_change_controller.go]: Sorry, I originally wrote errors.New because the text had no varargs in it. But now it does...

**Comment by mfojtik** [comment]
@ncdc updated

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
@ncdc the jenkins failure seems just flaky test-cmd, ok to merge?

**Comment by ncdc** [comment]
The delete all timeout is a known flake

**Comment by ncdc** [comment]
[merge]

**Pull request merged**